app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,73 @@
+class QuestionsController < ApplicationController
+  before_action :set_question, only: [:publish, :edit, :update, :destroy, :show]
+  before_action :validate_current_user_resource, only: [:publish, :edit, :update, :destroy]
+  skip_before_action :authorize, only: [:show]
+
+  def index
+    @q = Question.includes(:rich_text_content, :attachment_attachment).user_questions(current_user).chronological_order.ransack(params[:q])
+    @questions = @q.result
+  end
+
+  def new
+    @question = current_user.questions.build
+  end
+
+  def create
+    @question = current_user.questions.build(question_params)
+    @question.publish = true if params[:publish].present?
+    
+    if @question.save
+      message = @question.published_at? ? t('publish_success') : t('draft_success')
+      redirect_to questions_path, notice: message
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def publish
+    @question.publish = true
+
+    if @question.save
```

Comment:
First thing, this should work if we use, @question.update(publish: true)
Instead use some method on model though to publish